sklearn/metrics/tests/test_ranking.py

```diff
@@ -952,6 +952,36 @@ def test_alternative_lrap_implementation(n_samples, n_classes, random_state):
                n_classes, n_samples, random_state)
 
 
+def test_lrap_sample_weighting():
```

Comment:
This is tested by test_common.py unless I'm much mistaken. However it seems we don't generate any samples without labels there, perhaps intentionally since I think some metrics might not handle it